generators/entity-client/templates/react/src/main/webapp/app/entities/entity-dialog.tsx.ejs

```diff
@@ -139,7 +139,15 @@ export class <%= entityReactName %>Dialog extends React.Component<I<%= entityRea
     this.props.<%= val.action %>();
     <%_ }) _%>
   }
-
+  <%_ if (hasRelationshipQuery) { _%>
+  clearEmptyRelations = entity => {
+    Object.keys(entity).filter(k => entity[k] instanceof Object).forEach(k => {
+      if (entity[k]['id'] === '') {
+        delete entity[k];
+      }
+    });
+  }
+  <%_ } _%>
```

Comment:
Not the best way to do it, but it's pretty quick and does not add bloat to the code.